javaslang/src/test/java/javaslang/control/EitherTest.java

```diff
@@ -83,6 +84,17 @@ public void shouldFoldRight() {
     }
 
     @Test
+    public void shouldReturnSameWhenCallingMapOnLeft() {
+        final Either<Integer, Object> actual = Either.left(1);
+        assertThat(actual.map(Object::toString)).isSameAs(actual);
```

Comment:
is the method reference important here? If not, you could throw an exception, to make it clear that it's not called :)